### PR TITLE
wasm scope proposal

### PR DESCRIPTION
**Pull request by zuzunker**

### CODE REVIEW

**Review by a-hosani — APPROVED** [review]

**Review by a-hosani — APPROVED** [review]

**Review by a-hosani — APPROVED** [review]

File: packages/yoroi-extension/app/api/ada/lib/cardanoCrypto/rustLoader.js [hunk 1]
```diff
@@ -13,6 +17,113 @@ const MAX_TX_BYTES = 16384;
 
 type RustModuleLoadFlags = 'dontLoadMessagesSigning';
 
+function isWasmPointer(o: ?any): boolean {
+  return o != null && o.ptr != null && o.free != null;
+}
+
+function isPromise(o: ?any): boolean {
+  return o != null && (typeof o.then === 'function') && (typeof o.catch === 'function');
```
Comment (neuodev) [inline packages/yoroi-extension/app/api/ada/lib/cardanoCrypto/rustLoader.js]:
```suggestion
  return o instanceof Promise;
```

**Comment by neuodev** [comment]
I was trying to write some unit testing 

```js
   // rustLoader.test.js
    it('Test...', () => {
        const addr = 'addr1qf2w3pe8jsyvr9kutpv6e0rzuaym7dvq0rrz9699a7y3wwvz6g3gz764vxla692p4ttcekdw0smpedxuqq7a8t8sv3usrcv98z';
        const fixedAddr = RustModule.WasmScope(Module => bytesToHex(
          Module.WalletV4.Address.from_bech32(addr).to_bytes()
        ));

        console.log(fixedAddr)
    })
```

But I got the `Maximum call stack size exceeded` from the recursive function!

File: packages/yoroi-extension/app/api/ada/lib/cardanoCrypto/rustLoader.js (hunk 1 above)
Comment (zuzunker) [reply packages/yoroi-extension/app/api/ada/lib/cardanoCrypto/rustLoader.js]: Makes sense!

Comment (zuzunker) [reply packages/yoroi-extension/app/api/ada/lib/cardanoCrypto/rustLoader.js]: Removed the function completely now 👌

**Comment by zuzunker** [comment]
> But I got the `Maximum call stack size exceeded` from the recursive function!

@AhmedIbrahim336 , weird, call stack should not be happening. I have added some fixes, and also added a unit test now, plz check again: https://github.com/Emurgo/yoroi-frontend/pull/3093/commits/6056e156a8c02d9c8e83b00caf5cd4922bf48a55

UPD: latest version of the test is here https://github.com/Emurgo/yoroi-frontend/blob/c9a6df5413a679b051fff42d8431e1d8de02c9bb/packages/yoroi-extension/app/api/ada/lib/cardanoCrypto/rustLoader.test.js

**Comment by neuodev** [comment]
> @AhmedIbrahim336 , weird, call stack should not be happening. I have added some fixes, and also added a unit test now, plz check again: [6056e15](https://github.com/Emurgo/yoroi-frontend/commit/6056e156a8c02d9c8e83b00caf5cd4922bf48a55)

Nice, I will check it!

**Review by zuzunker — COMMENTED** [review]
/check

**Pull request merged**